### PR TITLE
Introduce ModulePath entity and extend Module interface

### PR DESCRIPTION
**Pull request by rexim**
Split #320

### CODE REVIEW

File: src/main/scala/me/rexim/morganey/module/Module.scala [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
+import scala.util._
+
+class Module(modulePath: ModulePath, classLoader: ClassLoader = Module.getClass.getClassLoader) {
+  def canonicalPath: String =
+    modulePath.asCanonicalPath.path
+
+  // TODO: Implement Module.bindings
+  def bindings: Try[Set[MorganeyBinding]] = ???
+
+  // TODO: Implement Module.dependencies
```
Comment (rexim) [inline src/main/scala/me/rexim/morganey/module/Module.scala]: I think these two subtasks can be merged into a single one. Their scope feels pretty small.

**Review by ForNeVeR — APPROVED** [review]

**Comment by rexim** [comment]
@ForNeVeR thanks for the review!

Initiation the protocol of subtask gathering!

**Comment by rexim** [comment]
The protocol has been completed.

**Comment by keddelzz** [comment]
I approve the changes, too!

Sorry for the late answer, I've been on vacation this weekend.

**Comment by rexim** [comment]
@keddelzz thanks for the review!

**Pull request merged**